smart-jump-ensime.el

```diff
@@ -0,0 +1,40 @@
+;;; smart-jump-ensime.el --- Register `ensime-mode' for `smart-jump'. -*- lexical-binding: t -*-
+
+;; Copyright (C) 2017 James Nguyen, Terje Larsen
+
+;; Author: James Nguyen <james@jojojames.com>, Terje Larsen <terlar@gmail.com>
+;; Maintainer: James Nguyen <james@jojojames.com>, Terje Larsen <terlar@gmail.com>
+;; URL: https://github.com/jojojames/smart-jump
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: emacs, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;; Register `ensime-mode' for `smart-jump'.
+
+;;; Code:
+(require 'smart-jump)
+(require 'ensime nil t)
+
+(defun smart-jump-ensime-register ()
+  "Register `ensime-mode' for `smart-jump'."
+  (smart-jump-register :modes 'ensime-mode
```

Comment:
I would've thought ensime's goto commands are async in nature.

Reply:
How to find out if it is async or not, if it was, would this setup not work? Because when I've tried it it was working.

I looked a bit into the [rabbit-hole](https://github.com/ensime/ensime-emacs/blob/34c7719c9739d1e5d55c3f776e8f32a417b4eea6/ensime-editor.el#L339) but couldn't really tell if it was async stuff or not.

Reply:
That looks like it might not be async. Usually async code has a few obvious callbacks in them.

We can leave it at that. If it works for you, lets go with it. If someone wants to tweak it later, they can do it. :)